### PR TITLE
Add TypeScript definitions

### PR DESCRIPTION
**Pull request by forabi**
Allows usage of this lib with TypeScript projects and auto completion and warnings in Visual Studio Code and other editors in JavaScript and TypeScript projects.

screenshot_20170605_033810
screenshot_20170605_033749

### CODE REVIEW

File: transformation-matrix.d.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+declare module 'transformation-matrix/scale' {
+  /** Calculate a scaling matrix */
+  export function scale(sx: number, sy: number): Matrix;
+}
+
+declare module 'transformation-matrix/shear' {
+  /** Calculate a shear matrix */
+  export function shear(shx: number, shy: number): Matrix;
+  export default { shear };
```
Comment (chrvadala) [inline transformation-matrix.d.ts]: is this really needed?

File: package.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
   "version": "1.3.1",
   "description": "2d transformation matrix functions written in ES6 syntax. Tree shaking ready!",
   "main": "./build-commonjs/index.js",
+  "typings": "transformation-matrix.d.ts",
```
Comment (chrvadala) [inline package.json]: add this file in the `files` array of the package.json 
```
"files": [
     "*.md",
     "build-*",
     "src",
     "tests"
   ],
```

File: transformation-matrix.d.ts (hunk 1 above)
Comment (chrvadala) [inline transformation-matrix.d.ts]: ` export default { shear };`

**Comment by jonpc2018** [comment]
Is it possible to include the additional parameters (i.e. cx, cy) on the rotate() method

From the transformation-matrix.d.ts file...
declare module 'transformation-matrix/rotate' {
+  /** Calculate a rotation matrix */
+  export function rotate(angle: number): Matrix;
+  /** Calculate a rotation matrix with a DEG angle */
+  export function rotateDEG(angle: number): Matrix;
+}

And the definition of the rotate() method
rotate(angle, [cx], [cy]) ⇒ Object
Calculate a rotation matrix

**Pull request merged**